cmd/tree-walk.go

```diff
@@ -169,11 +169,12 @@ func doTreeWalk(ctx context.Context, bucket, prefixDir, entryPrefixMatch, marker
 	if len(entries) == 0 {
 		return nil
 	}
+
 	for i, entry := range entries {
 		var leaf, leafDir bool
 
 		// Decision to do isLeaf check was pushed from listDir() to here.
-		if delayIsLeaf {
+		if delayIsLeaf && isLeaf != nil {
```

Comment:
this change is not correct, because it gives an impression to the reader that delayIsLeaf can be true and isLeaf can be nil (in which case the code will not strip the trailing `/` and not work as expected)

Hence it's better to leave it as before.

Or, do
```
if delayIsLeaf {
    if isLeaf == nil { return error }
```